### DISCUSSION TITLE
Hono/swagger-ui FULL configuration parameters available

### DISCUSSION THREAD

**Discussion by Luizfelz** [honojs/hono#3077] [Ideas]
Hello everyone!

It would be nice to have all the configuration parameters already settle up in Hono/swagger-ui. I mean, when using Hono/swagger-ui, we already have a bunch of configuration parameters from the original swagger-ui configuration as seen [here](https://github.com/swagger-api/swagger-ui/blob/0ce792c9d0965a8b6b5d75f5e1341ff6936a4cb0/docs/usage/configuration.md). However, some of them are missing in Hono/swagger-ui, and it would be really nice to have all the configurations available.

**Comment by yusukebe** [comment]
Hi @Luizfelz 

@naporin0624 @sor4chi Can you consider this matter if you want to do so!

**Comment by naporin0624** [comment]
Hi, @Luizfelz @yusukebe.

Since `hono/swagger-ui` serializes parameters and passes them to SwaggerUI, I will try to see if I can serialize the function properly.

It would be easier to implement if you know what is missing beforehand. I have checked on my end and found that these are the things that are missing. I will try to implement them.

- `queryConfigEnabled`
  - boolean
- `displayOperationId`
  - boolean
- `tagsSorter`
  - (a, b) => boolean
- `useUnsafeMarkdown`
  - boolean
- `onComplete`
  - Funciton
- `syntaxHighlight`
  - false | object
- `syntaxHighlight.activated`
  - boolean
- `syntaxHighlight.theme`
  - string[]
- `tryItOutEnabled`
  - boolean
- `requestSnippetsEnabled`
  - boolean
- `requestSnippets`
  - object
- `oauth2RedirectUrl`
  - string
- `showMutatedRequest`
  - boolean
- `request.curlOptions`
  - string[]
- `supportedSubmitMethods`
  - string[]
- `validatorUrl`
  - string
- `withCredentials`
  - boolean
- `modelPropertyMacro`
  - Funciton
- `parameterMacro`
  - Function

**Reply by Luizfelz** [reply]
Hi, @naporin0624

I'm sorry, I should have put the list with the missing parameters, as you did. But yes, that is almost all the parameters missing. You forgot only one in this list:

- ```persistAuthorization```
   - boolean

Thank you all for the reply. Hope I can use this parameters soon, haha. 😊

**Reply by naporin0624** [reply]
Oops, I had omitted to check! Thank you!

**Comment by naporin0624** [comment]
Created a PR.  I would be happy if you could check it out. I'm a little worried because I've never used these properties with swagger-ui.
https://github.com/honojs/middleware/pull/618

**Reply by Luizfelz** [reply]
Ok, I will check it. 😁

**Reply by AlphaLawless** [reply]
I will help you test

**Comment by Luizfelz** [comment]
As asked by @yusukebe [here](https://github.com/honojs/middleware/pull/618#issuecomment-2211474175), the PR review by @AlphaLawless  and I was replied [here](https://github.com/honojs/middleware/pull/618#issuecomment-2211497311)

**Reply by yusukebe** [reply]
@Luizfelz Thank! @naporin0624 Please check them.
